### PR TITLE
HttpCreds: Update app passwords url #5605

### PR DESCRIPTION
**Pull request by ckamm**
See also owncloud/core#27360

Create Account::makeServerVersion() along the way, to avoid
version literals like 0x0A0000.

### CODE REVIEW

File: src/gui/creds/httpcredentialsgui.cpp [hunk 1]
```diff
@@ -71,12 +71,18 @@ void HttpCredentialsGui::askFromUserAsync()
 QString HttpCredentialsGui::requestAppPasswordText(const Account* account)
 {
     if (account->serverVersionInt() < 0x090100) {
```
Comment (ogoffart) [inline src/gui/creds/httpcredentialsgui.cpp]: while you are at it, you can use makeServerVersion there too, then.

**Comment by ckamm** [comment]
@ogoffart I've tested againt 10.0.0 and the URL works. I'll make the other changes you suggest, maybe replacing our hex-version literals everywhere.

**Comment by ckamm** [comment]
@ogoffart Split into two commits, got rid of hex version literals, tested not-logged-in case: works. Approve?

**Comment by ogoffart** [comment]
So you tested with a version that were not logged, and after ligging, the browser show the page at the right location?

**Comment by ckamm** [comment]
@ogoffart Yes, first I get redirected to `index.php/login?redirect_url=%252Findex.php%252Fsettings%252Fpersonal%253Fsectionid%253Dsecurity#apppasswords` and then finally to `index.php/settings/personal?sectionid=security#apppasswords` which is good.

**Comment by ogoffart** [comment]
:+1:

**Pull request merged**